sys/net/routing/rpl/rpl.c

```diff
@@ -278,34 +278,35 @@ void *rpl_process(void *arg)
 
 void _rpl_update_routing_table(void)
 {
-    rpl_dodag_t *my_dodag = rpl_get_my_dodag();
-    rpl_routing_entry_t *rt;
+    rpl_dodag_t *my_dodag;
+    rpl_dodag_t *end;
+    rpl_routing_entry_t *rt = rpl_get_routing_table();
 
-    if (my_dodag != NULL) {
-        rt = rpl_get_routing_table();
+    for (uint8_t i = 0; i < rpl_max_routing_entries; i++) {
+        if (rt[i].used) {
+            if (rt[i].lifetime <= 1) {
+                memset(&rt[i], 0, sizeof(rt[i]));
+            }
+            else {
+                rt[i].lifetime = rt[i].lifetime - RPL_LIFETIME_STEP;
+            }
+        }
+    }
 
-        for (uint8_t i = 0; i < rpl_max_routing_entries; i++) {
-            if (rt[i].used) {
-                if (rt[i].lifetime <= 1) {
-                    memset(&rt[i], 0, sizeof(rt[i]));
+    for (my_dodag = rpl_dodags, end = my_dodag + RPL_MAX_DODAGS; my_dodag < end; my_dodag++) {
+        if (my_dodag->used) {
```

Comment:
Currently we do not really support a proper way of leaving a dodag. But this will change with subsequent PRs. The idea of this check is to ignore all dodags, which are stored in memory (as left-overs) and only handle the active ones.